src/sys/windows/selector.rs

```diff
@@ -159,7 +159,10 @@ impl SockState {
             self.poll_info.handles[0].status = 0;
             self.poll_info.handles[0].events = self.user_evts | afd::POLL_LOCAL_CLOSE;
 
-            let overlapped_ptr: PVOID = unsafe { transmute_copy(self_arc) };
+            // Increase the ref count as the memory will be used by the kernel.
+            let overlapped_ptr: *const Mutex<SockState> =
```

Comment:
This leaks Arc references because you're increasing the reference count twice now.
You have to decrease the ref count if self.afd.poll() fails and remove the forget(self_arc.clone()) call on success.

Reply:
You're right, I was completely focused on just the `transmute_copy` calls. I went a little overboard and moved the conversions into functions hopefully making the use a bit clearer.